torch/jit/_shape_functions.py

```diff
@@ -922,6 +922,46 @@ def native_batch_norm(input: List[int], weight: Optional[List[int]], bias: Optio
         _size = [0]
     return _copy(input), _size, _size
 
+def split_fn(input: List[int], split_sizes: List[int], dim: int = 0):
+    dim = maybe_wrap_dim(dim, len(input))
+    results: List[List[int]] = []
+    end_idx: List[int] = []
+    num_splits = len(split_sizes)
+    for i in range(num_splits):
+        if(i == 0):
```

Comment:
Nit: would you mind rewriting this to more closely follow how it's written in pytorch eager? It makes it easier to follow/mitigates bugs.

See: https://codebrowser.bddppq.com/pytorch/pytorch/aten/src/ATen/native/TensorShape.cpp.html#679

Reply:
Added. Please review.